Classes/Projection/Doctrine/AbstractDoctrineProjector.php

```diff
@@ -105,18 +122,64 @@ protected function remove($object)
      * @return void
      * @api
      */
-   public function reset()
-   {
-       $this->projectionPersistenceManager->drop($this->getReadModelClassName());
-   }
+    public function reset()
+    {
+        $this->projectionPersistenceManager->drop($this->getReadModelClassName());
+    }
 
-   /**
-    * If this projection is currently empty
-    *
-    * @return bool
-    */
-   public function isEmpty()
-   {
-       return $this->projectionPersistenceManager->count($this->getReadModelClassName()) === 0;
-   }
+    /**
+     * If this projection is currently empty
+     *
+     * @return bool
+     */
+    public function isEmpty()
+    {
+        return $this->projectionPersistenceManager->count($this->getReadModelClassName()) === 0;
+    }
+
+    /**
+     * Returns the last seen sequence number of events which has been applied to the concrete event listener.
+     *
+     * @return int
+     */
+    public function getHighestAppliedSequenceNumber(): int
+    {
+        $projectorIdentifier = $this->renderProjectorIdentifier(get_class($this));
+        $projectionState = $this->entityManager->find(ProjectionState::class, $projectorIdentifier);
+        return ($projectionState instanceof ProjectionState ? $projectionState->highestAppliedSequenceNumber : 0);
+    }
+
+    /**
+     * Saves the $sequenceNumber as the last seen sequence number of events which have been applied to the concrete
+     * event listener.
+     *
+     * @param int $sequenceNumber
+     * @return void
+     */
+    public function saveSequenceNumber(int $sequenceNumber)
+    {
+        $projectorIdentifier = $this->renderProjectorIdentifier(get_class($this));
+        $projectionState = $this->entityManager->find(ProjectionState::class, $projectorIdentifier);
+        if ($projectionState === null) {
+            $projectionState = new ProjectionState();
+            $projectionState->projectorIdentifier = $projectorIdentifier;
+        }
+        $projectionState->highestAppliedSequenceNumber = $sequenceNumber;
+        $this->entityManager->persist($projectionState);
+    }
+
+    /**
+     * Renders a projector identifier which can be used as an id in the projection state
+     *
+     * @param string $className
+     * @return string
+     */
+    private function renderProjectorIdentifier(string $className): string
```

Comment:
Why pass the class here? the `get_class($this)` call could just be in this method